coordinator/src/main/scala/filodb.coordinator/RowSource.scala

```diff
@@ -85,74 +75,86 @@ trait RowSource extends Actor with StrictLogging {
 
   def reading: Receive = {
     case GetMoreRows =>
-      val rows = (1 to rowsToRead).iterator
-                   .map(i => getNewRow())
-                   .takeWhile(_.isDefined)
-                   .toSeq.flatten
-      if (rows.nonEmpty) {
-        val maxSeqId = rows.map(_._1).max
-        currentHiSeqNo = Math.max(currentHiSeqNo, maxSeqId)
-        coordinatorActor ! NodeCoordinatorActor.IngestRows(dataset, version, rows.map(_._2), currentHiSeqNo)
+      if (batchIterator.hasNext) {
+        val (nextSeqId, nextBatch) = batchIterator.next
+        outstanding(nextSeqId) = nextBatch
+        coordinatorActor ! NodeCoordinatorActor.IngestRows(dataset, version, nextBatch, nextSeqId)
         // Go get more rows
-        if (currentHiSeqNo - lastAckedSeqNo < maxUnackedRows) {
+        if (outstanding.size < maxUnackedBatches) {
           self ! GetMoreRows
         } else {
-          logger.debug(s" ==> waiting: currentHi = $currentHiSeqNo, lastAcked = $lastAckedSeqNo")
-          context.system.scheduler.scheduleOnce(waitingPeriod, self, CheckCanIngest)
+          logger.debug(s" ==> waiting: outstanding seqIds = ${outstanding.keys}")
+          schedule(waitingPeriod, CheckCanIngest)
           context.become(waiting)
         }
       } else {
-        logger.debug(s" ==> doneReading: HiSeqNo = $currentHiSeqNo, lastAcked = $lastAckedSeqNo")
-        if (currentHiSeqNo == lastAckedSeqNo) sendFlush()
+        logger.info(s" ==> (${self.path.name}) doneReading: outstanding seqIds = ${outstanding.keys}")
+        if (outstanding.isEmpty)  { sendFlush() }
+        else {
+          schedule(waitingPeriod, CheckCanIngest)
+        }
```

Comment:
My current ScalaStyle settings have { } required after multiline if else.  I think this is a good idea in general because after code changes sometimes non obvious changes can lead to change of meaning.